data_store/table_extraction/process.py

```diff
@@ -75,38 +76,45 @@ def process(self, table: Table) -> None:
         :param table: The table to process.
         :return: None
         """
+
         self._strip_whitespace(table)
         self._lift_header(table)
         self._remove_merged_headers(table)
         self._drop_cols_by_name(table)
         self._remove_ignored_non_header_rows(table)
         self._replace_dropdown_placeholder(table)
         self._drop_bespoke_rows(table)
+
         if self.drop_empty_rows:
             self._drop_empty_rows(table)
+
         if self.drop_empty_tables and table.df.empty:
-            table.df = None
+            table.df = pd.DataFrame()
```

Comment:
`Table` class has table.db defined as `pd.DataFrame` type. You cannot make it `None`, unless you change the typing of the class property. `pd.DataFrame()` is how you represent an empty `DataFrame`. 

Tests needed updating from checking: 
`table.df is None` to `table.db.empty`
`table.df is not None` to `not table.db.empty`

Reply:
This change makes the attribute `drop_empty_tables` and the docstring `drop_empty_tables (bool, optional): Whether to drop tables that are entirely empty (default is False).` confusing.

Easier to update the type hint to `db: pd.DataFrame | None`?

Reply:
I don't completely understand this condition:
`if self.drop_empty_tables and table.df.empty` 

I was thinking that whenever `table.df.empty == True`. Doing `table.df = pd.DataFrame()` is redundant. 
Gave it a shot, I removed the whole if and tests still pass..
```
if self.drop_empty_tables and table.df.empty:
    table.df = pd.DataFrame()
```

So it is probably redundant as it's written and used now.

Not sure if making the `table.df` to be `pd.DataFrame | None` would be better. Would have me fixing 19 errors as a result.

Reply:
`drop_empty_tables` is never used. 

I am unsure how empty `df`'s are different from `None` `df`'s..

Reply:
May I delete the property?

Reply:
```
if self.drop_empty_tables and table.df.empty:
    table.df = pd.DataFrame()
```

Yes this is redundant, which is why it was doing `table.df = None` before. I guess without deep reading the code I would rather it stay as `table.df = None`.

If we _don't_ do that, then `drop_empty_tables` no longer makes sense as an attribute so we should change something there too.

Reply:
I removed it. Which does indeed change behaviour in theory. 
In practice, I cannot see the `drop_empty_tables` flag used anywhere.